### DISCUSSION TITLE
Push Notification and Job with Shiny

### DISCUSSION THREAD

**Discussion by tobus2012** [shinyorg/shiny#1362] [Q&A]
Good evening ,
I'm trying to use Shiny to check certain messages in the background and send push notifications.
I installed Shiny v2.1.0.2769, Shiny.Jobs 2.7.3, Shiny.Push 2.7.3, Shiny.Notification 2.7.3..

Then configured Shiny as shown in the code below:

    services.UseJobs(true);
    services.RegisterJob(new Shiny.Jobs.JobInfo(typeof(MyJob))
    {
        DeviceCharging = false,
        BatteryNotLow = false,
        RequiredInternetAccess = Shiny.Jobs.InternetAccess.None,
        Repeat = true ,
        PeriodicTime = new TimeSpan(0,0,15),
    });
    services.UseNotifications();
    services.UsePush<MyPushDelegate>();

Except that when I close the application the Job no longer works and yet I thought I read somewhere that even if the application is closed the job executes when it is launched by the operating system.

I missed something or did something wrong.

Concerning pushnotification I manage to generate the RegistrationToken which I use on FCM to send a message to my device, the first test failed but when I try again I do not receive a message again even though the Token is the same.

What doesn't work, FCM has limitations or constraints to respect.

Please help

**Comment by aritchie** [comment]
If you swipe away the app, jobs will shutdown.

You need to provide a reproducible sample and proper details in the issue template.  I simply don't have enough details to even make a guess.  Please note that I will not help for old versions.

**Comment by tobus2012** [comment]
Thank you for your feedback ok I understand that Shiny does not work when the application is closed.
Is there a possibility to do this??

  but I use the old version case I have a lot of dlls which run on netstandard 2.0 when I try to migrate to 2.1 I have errors that's why I am forced to use those which support netstandard 2.0.

Thank you for helping I am trying in vain to find an efficient approach to notify users for me the ideal solution was a job which when the conditions are met it launches and checks the new information or updated status of the current user and displays a local notification.

Attached the source codes
[PushNotificationShiny.zip](https://github.com/shinyorg/shiny/files/13735138/PushNotificationShiny.zip)

**Reply by aritchie** [reply]
Jobs will eventually launch when the platform decides, but Shiny can only obey the platform itself.  Push notifications will also generally kick the application awake.  You can view the platform docs for the way these things work.

> but I use the old version case I have a lot of dlls which run on netstandard 2.0 when I try to migrate to 2.1 I have errors that's why I am forced to use those which support netstandard 2.0.

I appreciate that, but this isn't a product.  I'm only willing to discuss latest versions and beyond.
